[hipfftw][docs] Documentation additions for hipFFTW (#2171)
ROCm/rocm-libraries: projects/hipfft/docs/conceptual/overview.rst

application and the backend FFT library, marshalling inputs into the
backend and results back to the application.

hipFFTW exports an interface that is a subset of FFTW_'s interface. Applications using

> the latter for computing discrete Fourier transforms (on CPUs) may use hipFFTW as a

Reviewer: ```suggestion
the latter for computing discrete Fourier transforms (on CPUs) can use hipFFTW as a
```

Author: Done in bb1f53d33dc9ed13155ece673ffae750bd1e2c2b.

> drop-in substitution that would leverage GPU computations instead with minimal

Reviewer: ```suggestion
drop-in substitution to leverage GPU computations instead with minimal
```

Author: Done in bb1f53d33dc9ed13155ece673ffae750bd1e2c2b.

modifications to their source code. Namely, it is recommended to replace

> inclusions of ``fftw3.h`` by inclusions of ``hipfft/hipfftw.h`` and link

Reviewer: ```suggestion
inclusions of ``fftw3.h`` with inclusions of ``hipfft/hipfftw.h`` and link
```

Author: Done in bb1f53d33dc9ed13155ece673ffae750bd1e2c2b.

with hipFFTW instead of fftw3 or fftw3f. For more details, see the
:doc:`hipFFTW API reference <../reference/hipfftw-api-usage>`.

.. note::
   hipFFTW supports single- and double-precision calculations; long-double and

>    quad-precision floating point arithmetics are not supported.

Reviewer: ```suggestion
   hipFFTW supports single- and double-precision calculations; long-double and
   quad-precision floating point operations are not supported.
```

Author: Done in bb1f53d33dc9ed13155ece673ffae750bd1e2c2b.
